kml.cc

```diff
@@ -406,34 +406,28 @@ void wpt_icon(xg_string args, const QXmlStreamAttributes*)
 
 void trk_coord(xg_string args, const QXmlStreamAttributes*)
 {
-  int consumed = 0;
-  double lat, lon, alt;
-  Waypoint* trkpt;
-  int n = 0;
-
   route_head* trk_head = route_head_alloc();
-
-  QString iargs = args;
   if (wpt_tmp && !wpt_tmp->shortname.isEmpty()) {
     trk_head->rte_name  = wpt_tmp->shortname;
   }
   track_add_head(trk_head);
-  while ((n = sscanf(CSTRc(iargs), "%lf,%lf,%lf%n", &lon, &lat, &alt, &consumed)) > 0) {
-    trkpt = new Waypoint;
-    trkpt->latitude = lat;
-    trkpt->longitude = lon;
-
-    // Line malformed or two-arg format without alt .  Rescan.
-    if (2 == n) {
-      sscanf(CSTRc(iargs), "%lf,%lf%n", &lon, &lat, &consumed);
-    }
 
-    if (3 == n) {
-      trkpt->altitude = alt;
-    }
+  auto vecs = args.simplified().split(' ');
+  for(const auto& vec : vecs) {
+    QStringList coords = vec.split(',');
```

Comment:
I recommend making Qt containers const when possible as it makes it clear you don't want to modify the container and  it prevents issues with detach.

Reply:
Fixed as above.